### ISSUE TITLE
Error in lazy decomposition when scaling signal to normalize Poissonian noise

### ISSUE THREAD

**Issue by hakonanes** [hyperspy/hyperspy#2109]
Running

```python
>>> import hyperspy.api as hs
>>> s = hs.load(diffraction_patterns, lazy=True)
>>> s.change_dtype('float')
>>> s.decomposition(algorithm='PCA', output_dimension=200, normalize_poissonian_noise=True)
```
throws

```python
Traceback (most recent call last):

  File "<ipython-input-1-88d7158fb358>", line 32, in <module>
    normalize_poissonian_noise=poisson_noise)

  File "/home/hakon/miniconda3/envs/pxm/lib/python3.6/site-packages/hyperspy/_signals/lazy.py", line 764, in decomposition
    data.sum(axis=range(ndim)),

  File "/home/hakon/miniconda3/envs/pxm/lib/python3.6/site-packages/dask/array/core.py", line 1787, in sum
    split_every=split_every, out=out)

  File "/home/hakon/miniconda3/envs/pxm/lib/python3.6/site-packages/dask/array/reductions.py", line 232, in sum
    dtype=dt, split_every=split_every, out=out)

  File "/home/hakon/miniconda3/envs/pxm/lib/python3.6/site-packages/dask/array/reductions.py", line 130, in reduction
    axis = validate_axis(axis, x.ndim)

  File "/home/hakon/miniconda3/envs/pxm/lib/python3.6/site-packages/dask/array/utils.py", line 144, in validate_axis
    raise TypeError("Axis value must be an integer, got %s" % axis)

TypeError: Axis value must be an integer, got range(0, 2)
```

The issue is the specification of axes in `/hyperspy/_signals/lazy.py` lines 764 and 765:

```python
bH, aG = da.compute(
    data.sum(axis=range(ndim)),
    data.sum(axis=range(ndim, ndim + sdim)))
```

which I guess (!) should be

```python
bH, aG = da.compute(
    data.sum(axis=(0, ndim - 1)),
    data.sum(axis=(ndim, ndim + sdim - 1)))
```

**Comment by francisco-dlp** [comment]
Good catch! Since you've found the solution, wouldn't you like to send a pull request to fix it?

**Comment by hakonanes** [comment]
The suggested change will work fine for 4D datasets, but how about higher dimensional data? I have not considered this.

**Comment by francisco-dlp** [comment]
It seems to me that the real issue is that the code is using Python2 syntax that haven't been updated to Python3. Could you test if replacing

```python
bH, aG = da.compute(
    data.sum(axis=range(ndim)),
    data.sum(axis=range(ndim, ndim + sdim)))
```

with

```python
bH, aG = da.compute(
    data.sum(axis=tuple(range(ndim))),
    data.sum(axis=tuple(range(ndim, ndim + sdim))))
```
fixes the issue?

**Comment by hakonanes** [comment]
It does! Made a PR with those two lines changed [here](https://github.com/hyperspy/hyperspy/pull/2111).

**Comment by francisco-dlp** [comment]
Fixed in #2111
